### ISSUE TITLE
Stuck on `Error: Gone`

### ISSUE THREAD

**Issue by patrickberkeley** [stephenplusplus/gcs-resumable-upload#15]
I'm deploying an Ember application to a GCS using the [ember-cli-deploy-gcloud-storage](https://github.com/knownasilya/ember-cli-deploy-gcloud-storage) and [ember-cli-deploy-gcs-index](https://github.com/envoy/ember-cli-deploy-gcs-index) addons.

Everything was working correctly when I used a GC service account with Project Owner permissions, but this seemed like overkill so I was experimenting trying different service accounts with lesser permissions. Somewhere along the line, deploying the ember app went from returning `Error: Unauthorized` to `Error: Gone`. Now no matter what I try, the bucket seems to be stuck in the state of trying to resume a failed upload. Below is the console output from running `ember deploy`.

I've tried:

1. Uploading directly through the GC Storage UI the files I thought were missing. The goal being to trick the bucket into thinking everything was uploaded.
1. Deleting everything from the bucket.

Neither of these worked, so I've resorted to destroying the bucket completely and starting fresh.

My question: Is there a way to clear the partially uploaded file cache on a bucket (or clear wherever it keeps track of uploads) without destroying the entire bucket?

```sh
- Error: Gone
- Error: Gone
    at Request._callback (/Users/uname/Code/union/portal/node_modules/ember-cli-deploy-gcloud-storage/node_modules/google-cloud/node_modules/@google-cloud/storage/node_modules/gcs-resumable-upload/index.js:274:25)
    at Request.self.callback (/Users/uname/Code/union/portal/node_modules/request/request.js:187:22)
    at emitTwo (events.js:100:13)
    at Request.emit (events.js:185:7)
    at Request.<anonymous> (/Users/uname/Code/union/portal/node_modules/request/request.js:1044:10)
    at emitOne (events.js:90:13)
    at Request.emit (events.js:182:7)
    at IncomingMessage.<anonymous> (/Users/uname/Code/union/portal/node_modules/request/request.js:965:12)
    at emitNone (events.js:85:20)
    at IncomingMessage.emit (events.js:179:7)
```

**Comment by stephenplusplus** [comment]
@capstan does this sound like the issue brought up recently about explicitly canceling/deleting a resumable upload, or is this something else?

**Comment by Capstan** [comment]
`410 Gone` is related to poisoned or expired upload sessions. Upload sessions expire 7 days after creation, but may linger for a bit before being garbage collected. Session poisoning happens if we get corrupt data (i.e., your provided hash doesn't match the data you sent) or the service experienced a small class of unrecoverable errors and likely tried to return a 503. Unfortunately, if a session is `Gone`, it's gone and a new upload session must be erected.

**Comment by stephenplusplus** [comment]
Thanks for the info! I've put a PR together at #16. @patrickberkeley would you mind giving it a shot?

**Comment by Capstan** [comment]
I erred in my comment; it does *not* include expired sessions, which will return `404 Not Found`. This is purely for cases of bad data transmission or service-side error rendering the session unrecoverable. We call them "terminated" sessions.

**Comment by patrickberkeley** [comment]
@stephenplusplus thanks for following up on this and for the fix. Unfortunately I don't have a solid way to reproduce the error. The bucket initial got into the error'ed state when I was changing permissions, but the only thing I was doing from the command line was deploying. I think the `Gone` state was just a fluke.

Let me know there are steps I can take to reproduce and I'd be happy to give them a shot.

**Comment by stephenplusplus** [comment]
No problem, I'm not sure how to reproduce either. I think I'll just make a new release which *should* help if someone else winds up in situation you were in. If it doesn't, we'll just keep working through it :)

**Comment by stephenplusplus** [comment]
Published under `0.7.4`. Thanks for reporting @patrickberkeley!
